buffer.go

```diff
@@ -109,39 +109,39 @@ func (b *buffer) readNext(need int) ([]byte, error) {
 // If possible, a slice from the existing buffer is returned.
 // Otherwise a bigger buffer is made.
 // Only one buffer (total) can be used at a time.
-func (b *buffer) takeBuffer(length int) []byte {
+func (b *buffer) takeBuffer(length int) ([]byte, error) {
 	if b.length > 0 {
-		return nil
+		return nil, ErrUnreadTxRows
```

Comment:
Please do not return the error here. Leave the buffer unchanged.
This specific error should only be returned when we know that it is this specific error.